cmd/release-controller-api/http.go

```diff
@@ -1477,6 +1477,37 @@ func (c *Controller) httpReleaseLatestDownload(w http.ResponseWriter, req *http.
 	http.Redirect(w, req, u, http.StatusFound)
 }
 
+// Find the stream from app.ci and check whether it has the
+// inconsistency annotation
+func (c *Controller) doesInconsistencyExist(tag string) bool {
+	releaseStream, err := c.getStreamFromTag(tag)
+	if err == nil {
+		if inconsistencyMessage, ok := releaseStream.Annotations[releasecontroller.ReleaseAnnotationInconsistency]; ok {
+			_, nil := jsonArrayToString(inconsistencyMessage)
+			if nil == nil {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func (c *Controller) tableLink(config *releasecontroller.ReleaseConfig, tag imagev1.TagReference) string {
+	if canLink(tag) {
+		if value, ok := tag.Annotations[releasecontroller.ReleaseAnnotationKeep]; ok {
+			return fmt.Sprintf(`<td class="text-monospace"><a title="%s" class="%s" href="/releasestream/%s/release/%s">%s <span>*</span></a></td>`, template.HTMLEscapeString(value), phaseAlert(tag), template.HTMLEscapeString(config.Name), template.HTMLEscapeString(tag.Name), template.HTMLEscapeString(tag.Name))
+		}
+		if strings.Contains(tag.Name, "nightly") && c.doesInconsistencyExist(tag.Name) {
```

Comment:
The `doesInconsistencyExist` method is called for every stream. That might not be ideal (it will subsequently call `getStreamFromTag`, which lists the imagestreams every time).
Can you perhaps fetch the imagestreams once in the `httpReleases` release method, and use that data to render the page, rather than calling `c.releaseLister.List(labels.Everything()` for every stream?

Reply:
In python, `strings.Contains(tag.Name, "nightly")` is `False` then the check after `&&` is not executed. Is it the same for Go?

Reply:
I think its called short circuiting.

Not sure if we are talking about different things now haha?

Reply:
Afaik, the same applies to golang, if the first statement is evaluated as false, the second one will not execute. The `doesInconsistencyExist` method will be called for every nightly stream (too many times).  You don't need to `c.releaseLister.List(labels.Everything()` separately for every nightly, but rather do it once and use the data subsequently.

Reply:
Afaik, the same applies to golang, if the first statement is evaluated as false, the second one will not execute. The `doesInconsistencyExist` method will be called for every nightly stream (too many times).  You don't need to `c.releaseLister.List(labels.Everything()` separately for every nightly, but rather do it once and use the data subsequently.

Reply:
Ahh! Makes sense. Will work on the change.

Reply:
Not sure if this is what you envisioned https://github.com/openshift/release-controller/pull/577/commits/243fe90b2e3b25003091ca657cdd55dc31b0ac4d